From 10d8f3f63052e11da4275f9dd0ad2ef51bb28f93 Mon Sep 17 00:00:00 2001
From: Darren Etheridge <detheridge@ti.com>
Date: Fri, 21 Jun 2013 13:52:27 -0500
Subject: [PATCH 2/4] drm/tilcdc fixing i2c/slave initialization race

In certain senarios drm will initialize before i2c this means that i2c
slave devices like the nxp tda998x will fail to be probed.  This patch
detects this condition then defers the probe of the slave device and
the tilcdc main driver.

Signed-off-by: Darren Etheridge <detheridge@ti.com>
Acked-by: Rob Clark <robdclark@gmail.com>
Signed-off-by: Dave Airlie <airlied@redhat.com>
---
 drivers/gpu/drm/tilcdc/tilcdc_slave.c |    2 --
 1 file changed, 2 deletions(-)

Index: linux-3.12.24-rt38-r7s8/drivers/gpu/drm/tilcdc/tilcdc_slave.c
===================================================================
--- linux-3.12.24-rt38-r7s8.orig/drivers/gpu/drm/tilcdc/tilcdc_slave.c
+++ linux-3.12.24-rt38-r7s8/drivers/gpu/drm/tilcdc/tilcdc_slave.c
@@ -338,8 +338,6 @@ static int slave_probe(struct platform_d
 		return ret;
 	}
 
-	mod->preferred_bpp = slave_info.bpp;
-
 	i2c_node = of_find_node_by_phandle(i2c_phandle);
 	if (!i2c_node) {
 		dev_err(&pdev->dev, "could not get i2c bus node\n");
